### ISSUE TITLE
XCP-NG 8.3 kernel loading issue in EFI mode with iPXE + multiboot2 patches

### ISSUE THREAD

**Issue by netmax79** [xcp-ng/xcp#663]
Hi,

i just tried to invenst some time to get XCP-NG installer booted in EFI mode using iPXE + multiboot2 patches.

It's working fine for XCP-NG 8.2.1 but not for the new 8.3.0 release. It looks like the kernel behaves somewhat different. Maybe there were some changes in the build process?

The resulting error message from iPXE is:

> ipxe: Could not boot: Requested memory not available (https://ipxe.org/46038185)

**Comment by olivierlambert** [comment]
Just a thought: could you try to use iPXE with debug mode in case it provides more info?

**Comment by netmax79** [comment]
That task is already on my list, i will report back results once i tried that.

**Comment by netmax79** [comment]
After some more testing, it seems that this is only happening on my test VM which is running on an XCP-NG 8.2 Host.
I tried to boot the same iPXE stack physical hardware (DELL R660) and that seems to work without any issue.

The problem is still a bit weird because the older installer (8.2.1) boots just fine also in a VM.

**Comment by olivierlambert** [comment]
That's weird indeed, but getting working on real hardware is the most important thing and a good news in the end.

**Comment by stormi** [comment]
How do you get the VM to boot on your patched iPXE?

**Comment by netmax79** [comment]
Our dhcp server is presenting my iPXE by ftp to the VM/physical machines. It loads a menu structure which is dynamically created from single .boot files. One of it for XCP 8.3.0 has the following content:

`console
console --picture http://http.server/.ks/boot-xcpng.png --x 800 --y 600 --left 32 --right 32 --top 32 --bottom 48

set base-url http://http.server/xcp-ng/8.3.0

kernel ${base-url}/boot/xen.gz
module ${base-url}/boot/vmlinuz
module ${base-url}/install.img
imgargs xen.gz dom0_max_vcpus=1-16 dom0_mem=max:8192M com1=115200,8n1 console=com1,vga
imgargs vmlinuz xencons=hvc console=hvc0 console=tty0 device_mapper_multipath=enabled answerfile=http://http.server/.ks/xcp-ng83.cfg
boot`

**Comment by stormi** [comment]
Does the refreshed 8.2.1 installer also boot fine?

**Comment by netmax79** [comment]
I was already testing with the updated 8.2.1 installer. It worked fine.

**Comment by stormi** [comment]
Adding @ydirson to the conversation.

**Comment by Fredouye** [comment]
Hi

same here in nested mode, with an ESXi 8.0 U3b running on a Xeon E5-2470 v2 CPU.

I cannot boot the 8.3.0 installer, either in BIOS or UEFI mode. It starts booting, and the VM is restarted after a few seconds.

**Comment by stormi** [comment]
@Fredouye Are you using iPXE to boot?

**Comment by Razor648** [comment]
i have the same issue in nested XCP-ng in VMware when i mounted iSO in my Dom0 VM. Don't boot in UEFI.

**Comment by stormi** [comment]
@Razor648 This issue is about booting using iPXE, but you are mentioning an ISO. Could it be a different issue?

**Comment by Razor648** [comment]
Yes it's on ISO. UEFI or Legacy BIOS. It's different issue, indeed.
